src/app/search/SearchBar.tsx

```diff
@@ -148,23 +157,51 @@ const Row = React.memo(
       );
     }
 
+    const rowContents = () => {
+      switch (item.type) {
+        case SearchItemType.Help:
+          return t('Header.FilterHelpMenuItem');
+        case SearchItemType.ArmoryEntry:
+          return (
+            <>
+              {item.armoryItem!.name}
```

Comment:
There's some type trickery with a union you can do to get TS to understand that when `item.type === SearchItemType.ArmoryEntry` then `armoryItem` is present.

Reply:
It works but the type definition is a bit of an abomination [`6c9c163` (#9045)](https://github.com/DestinyItemManager/DIM/pull/9045/commits/6c9c1633a991bf76469b294cbf7d573aeb896c44#diff-ba1137a0099f6f059f352395fc7cfa15e309d73dfc27681b3266f5329c0a3780R55-R60) -- it doesn't seem to work without the `Exclude` because TS sees that `BaseSearchItem` could totally exist with `type` set to `SearchItemType.ArmoryEntry` and complains the property can't be found on `BaseSearchItem`.

Reply:
Yeah that's what I had in mind but it really is awkward